test/IdentityServer4.RavenDB.Storage.Tests/DocumentStoreHolderTests/DocumentStoreHolderTests.cs

```diff
@@ -1,128 +1,117 @@
-
-            using (GetOperationalDocumentStoreHolder(store))
-            {
-            } 
-            
-            AssertCertificateDisposed(store);
-        }
-
-        private X509Certificate2 GetCertificate()
-        {
-            const string certificatePath =
-                "IdentityServer4.RavenDB.IntegrationTests.DocumentStoreHolderTests.certificate.txt";
-
-            var cert = EmbeddedResourceHelper.GetFileContent(certificatePath);
-
-            var bytes = Convert.FromBase64String(cert);
-            
-            return new X509Certificate2(bytes);
-        }
-
-        private DocumentStore ConfigureDocumentStore(X509Certificate2 certificate = null)
-        {
-            return DocumentStoreHelper.InitializeDocumentStore(store =>
-            {
-                store.Database = "some-database-name";
-                store.Urls = new[]
-                {
-                    "https://127.0.0.1"
-                };
-                store.Certificate = certificate;
-            });
-        }
-
-        private ConfigurationDocumentStoreHolder GetConfigurationDocumentStoreHolder(IDocumentStore store)
-        {
-            return new ConfigurationDocumentStoreHolder(store);
-        }
-        
-        private OperationalDocumentStoreHolder GetOperationalDocumentStoreHolder(IDocumentStore store)
-        {
-            return new OperationalDocumentStoreHolder(store);
-        }
-
-        private void AssertDocumentStoreDisposed(IDocumentStore store)
-        {
-            Assert.True(store.WasDisposed);
-        }
-
-        private void AssertDocumentStoreNotDisposed(IDocumentStore store)
-        {
-            Assert.False(store.WasDisposed);
-        }
-
-        private void AssertCertificateCreated(IDocumentStore store)
-        {
-            Assert.NotNull(store.Certificate.RawData);
-            Assert.NotNull(store.Certificate.PublicKey);
-            Assert.NotNull(store.Certificate.Thumbprint);
-        }
-
-        private void AssertCertificateDisposed(DocumentStore store)
-        {
-            Assert.Throws<CryptographicException>(() => store.Certificate.RawData);
-            Assert.Throws<CryptographicException>(() => store.Certificate.PublicKey);
-            Assert.Throws<CryptographicException>(() => store.Certificate.Thumbprint);
-        }
-    }
+﻿using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using IdentityServer4.RavenDB.Storage.DocumentStoreHolder;
+using IdentityServer4.RavenDB.Storage.Helpers;
+using IdentityServer4.RavenDB.Storage.Options;
+using Raven.Client.Documents;
+using Xunit;
+
+namespace IdentityServer4.RavenDB.Storage.Tests.DocumentStoreHolderTests
+{
+    public class DocumentStoreHolderTests : IntegrationTestBase
+    {
+        [Fact]
+        public void DocumentStore_IsDisposed_WhenConfigurationDocumentStoreHolderIsDisposed()
+        {
+            var options = GetRavenDbConfigurationStoreOptions();

```

Comment:
If you don't modify the `options` as part of the test, then you can encapsulate them in the `GetConfigurationDocumentStoreHolder()` method.

Reply:
👍